src/http.cpp

```diff
@@ -684,21 +665,20 @@ m_off_t SpeedController::calculateSpeed(long long numBytes)
 
     while (transferBytes.size())
     {
-        dstime deltaTime = currentTime - transferTimes.front();
-        if (deltaTime <= SPEED_MEAN_INTERVAL_DS)
+        map<dstime, m_off_t>::iterator it = transferBytes.begin();
+        dstime deltaTime = currentTime - it->first;
+        if (deltaTime < SPEED_MEAN_INTERVAL_DS)
```

Comment:
Yes, if `SPEED_MEAN_INTERVAL_DS'  is one, for example, only the element corresponding to the current decisecond should be kept in the map. Otherwise two deciseconds would be taken into account.